### PR TITLE
Remove checking if an unsigned value is less than zero

### PR DESCRIPTION
**Pull request by robertmaynard**
`result_cache::has_result` would try and see if a unsigned value was less than zero, a state that it can never be.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

**Comment by robertmaynard** [comment]
@gpucibot merge

**Comment by harrism** [comment]
@robertmaynard I realize this is a trivial PR, but just so you know we do require two cpp-codeowners approvals on libcudf PRs (can't be enforced by github, unfortunately).

**Comment by robertmaynard** [comment]
> @robertmaynard I realize this is a trivial PR, but just so you know we do require two cpp-codeowners approvals on libcudf PRs (can't be enforced by github, unfortunately).

:+1:  Will do going forward

**Pull request merged**
